driver-core/src/main/com/mongodb/internal/connection/SocketStreamHelper.java

```diff
@@ -81,11 +82,12 @@ static void initialize(final Socket socket, final InetSocketAddress inetSocketAd
     private static void setExtendedSocketOptions(final Socket socket) {
         if (Arrays.stream(ExtendedSocketOptions.class.getDeclaredFields()).anyMatch(f -> f.getName().equals(TCP_KEEPCOUNT))) {
             try {
-                socket.setOption((SocketOption<Integer>) ExtendedSocketOptions.class.getField(TCP_KEEPCOUNT).get(null),
+                Method setOptionMethod = Socket.class.getMethod("setOption", SocketOption.class, Object.class);
+                setOptionMethod.invoke(socket, ExtendedSocketOptions.class.getDeclaredField(TCP_KEEPCOUNT).get(null),
                         TCP_KEEPCOUNT_LIMIT);
-                socket.setOption((SocketOption<Integer>) ExtendedSocketOptions.class.getField(TCP_KEEPIDLE).get(null),
+                setOptionMethod.invoke(socket, ExtendedSocketOptions.class.getDeclaredField(TCP_KEEPCOUNT).get(null),
```

Comment:
Copy pasta (as Ross would say): Use all three field names

Reply:
Drat! I now use all three names.

Reply:
This makes me think we should add a test.  I think it would be easy to add a unit test that asserts on these values via calls to `Socket.getOption()`.  Please see if you can add one.

Reply:
There is a test in `SocketStreamHelperSpecification` that checks these added socket options. I have updated the test to call `getOption` reflectively.